Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/FRFSTicketService.hs

```diff
@@ -629,3 +634,154 @@ getFrfsConfig (pId, mId) opCity = do
   let isEventOngoing' = fromMaybe False isEventOngoing
       ticketsBookedInEvent = fromMaybe 0 ((.ticketsBookedInEvent) =<< stats)
   return FRFSTicketService.FRFSConfigAPIRes {isEventOngoing = isEventOngoing', ..}
+
+data StationResult = StationResult
+  { code :: Kernel.Prelude.Text,
+    name :: Kernel.Prelude.Text,
+    vehicleType :: Spec.VehicleCategory,
+    lat :: Double,
+    lon :: Double
+  }
+
+instance HasCoordinates StationResult where
+  getCoordinates stop = LatLong (stop.lat) (stop.lon)
+
+getFrfsAutocomplete ::
+  ( Kernel.Prelude.Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person),
+    Kernel.Types.Id.Id Domain.Types.Merchant.Merchant
+  ) ->
+  Kernel.Prelude.Maybe Text ->
+  Context.City ->
+  Kernel.External.Maps.Types.LatLong ->
+  BecknV2.FRFS.Enums.VehicleCategory ->
+  Environment.Flow [API.Types.UI.FRFSTicketService.AutocompleteRes]
+getFrfsAutocomplete (_, mId) text opCity origin vehicle = do
+  merchantOpCity <- CQMOC.findByMerchantIdAndCity mId opCity >>= fromMaybeM (MerchantOperatingCityNotFound $ "merchant-Id-" <> mId.getId <> " ,city: " <> show opCity)
+  frfsConfig <-
+    CQFRFSConfig.findByMerchantOperatingCityId merchantOpCity.id
+      >>= fromMaybeM (InternalError $ "FRFS config not found for merchant operating city Id " <> show merchantOpCity.id)
+  case text of
+    Nothing -> do
+      allStops <- QStation.getTicketPlacesByMerchantOperatingCityIdAndVehicleType merchantOpCity.id vehicle
+
+      currentTime <- getCurrentTime
+
+      let serviceableStops = DTB.findBoundedDomain allStops currentTime
+
+      let filteredStops =
+            filter
+              ( \stop ->
+                  let straightLineDist = highPrecMetersToMeters (CD.distanceBetweenInMeters origin (LatLong (fromMaybe merchantOpCity.lat stop.lat) (fromMaybe merchantOpCity.long stop.lon)))
+                   in straightLineDist <= fromMaybe (Meters (maxBound :: Int)) frfsConfig.straightLineDistance
```

Comment:
If Domain type of `frfsConfig.straightLineDistance` becomes Non Maybe then fromMaybe can be removed.